src/connectivity.jl

```diff
@@ -283,6 +283,204 @@ function strongly_connected_components end
     return components
 end
 
+"""
+    strongly_connected_components_kosaraju(g)
+
+Compute the strongly connected components of a directed graph `g` using Kosaraju's Algorithm. 
+(https://en.wikipedia.org/wiki/Kosaraju%27s_algorithm).
+
+Return an array of arrays, each of which is the entire connected component.
+
+### Performance
+Time Complexity : O(|E|+|V|) 
+|V| = Number of vertices
+|E| = Number of edges
+This algorithm requires 2 depth first search passes. Also the algorithm reverses the graph once after one dfs.
+At the end, the graph is reversed again to restore the original graph.
+Also, an iterative version of dfs is written so that this graph doesn't show "stackoverflow" when run on large 
+graphs.
+
+Space Complexity : O(|V|) 
+This memory is used for vectors named color, original, component and components. This space complexity 
+excludes the space required to store the graph.
+
+### Examples
+```jldoctest
+
+julia> g=SimpleDiGraph(3)
+{3, 0} directed simple Int64 graph
+
+julia> edge_list=[(1,2),(2,3)]
+2-element Array{Tuple{Int64,Int64},1}:
+ (1, 2)
+ (2, 3)
+
+julia> for e in edge_list
+           add_edge!(g, e[1], e[2])
+       end
+
+julia> strongly_connected_components_kosaraju(g)
+3-element Array{Array{Int64,1},1}:
+ [1]
+ [2]
+ [3]
+
+
+julia> g=SimpleDiGraph(3)
+{3, 0} directed simple Int64 graph
+
+julia> edge_list=[(1,2),(2,3),(3,1)]
+3-element Array{Tuple{Int64,Int64},1}:
+ (1, 2)
+ (2, 3)
+ (3, 1)
+
+julia> for e in edge_list
+           add_edge!(g, e[1], e[2])
+       end
+
+julia> strongly_connected_components_kosaraju(g)
+1-element Array{Array{Int64,1},1}:
+ [2, 3, 1]
+
+
+julia> g=SimpleDiGraph(3)
+{3, 0} directed simple Int64 graph
+
+julia> edge_list=[(1,2),(2,3),(3,2)]
+3-element Array{Tuple{Int64,Int64},1}:
+ (1, 2)
+ (2, 3)
+ (3, 2)
+
+julia> for e in edge_list
+           add_edge!(g, e[1], e[2])
+       end
+
+
+julia> g=SimpleDiGraph(11)
+{11, 0} directed simple Int64 graph
+
+julia> edge_list=[(1,2),(2,3),(3,4),(4,1),(3,5),(5,6),(6,7),(7,5),(5,8),(8,9),(9,8),(10,11),(11,10)]
+13-element Array{Tuple{Int64,Int64},1}:
+ (1, 2)  
+ (2, 3)  
+ (3, 4)  
+ (4, 1)  
+ (3, 5)  
+ (5, 6)  
+ (6, 7)  
+ (7, 5)  
+ (5, 8)  
+ (8, 9)  
+ (9, 8)  
+ (10, 11)
+ (11, 10)
+
+julia> for e in edge_list
+           add_edge!(g, e[1], e[2])
+       end
+
+julia> strongly_connected_components_kosaraju(g)
+4-element Array{Array{Int64,1},1}:
+ [11, 10]    
+ [2, 3, 4, 1]
+ [6, 7, 5]   
+ [9, 8]      
+
+```
+"""
+
+function strongly_connected_components_kosaraju end
+@traitfn function strongly_connected_components_kosaraju(g::AG::IsDirected) where {T, AG <: AbstractGraph{T}}
+       
+   nvg = nv(g)    
+
+   components = Vector{Vector{T}}()    # Maintains a list of strongly connected components
+   sizehint!(components, nvg)
```

Comment:
I am not sure that this `sizehint` here is very wise. The problem is, that `sizehint!` usually does allocate a vector of the required size in the background. If we only have a few strongly connected components, this allocates more memory than necessary.

Reply:
Removed `sizehint` .